### PR TITLE
Created ad-hoc signed jar for testing filtering of jar file

### LINKED ISSUE #40156: Generate ad-hoc signed JAR in `JarResultBuildStepTest`
In #40001 we're using the dependency plugin to download an already signed JAR to run the `JarResultBuildStepTest`. It would be nice if we could create one on the fly if possible (with Shrinkwrap?) and then sign it with some randomly generated keys.

_Originally posted by @galderz in https://github.com/quarkusio/quarkus/issues/40001#issuecomment-2058792934_

### PR DESCRIPTION
**Pull request by DShivansh**
- Fixes #40156

### CODE REVIEW

File: core/deployment/pom.xml [hunk 1]
```diff
@@ -147,34 +147,19 @@
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk18on</artifactId>
```
Comment (gastaldi) [inline core/deployment/pom.xml:152]: They are both missing the `<scope>test</scope>`

File: core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java [hunk 2]
```diff
@@ -2,27 +2,64 @@
...
+        Path signedJarPath = tmp.resolve("signed-jar.jar");
+        archive.as(ZipExporter.class).exportTo(new File(unsignedJarPath.toUri()), true);
+        JarSigner signer = new JarSigner.Builder(createPrivateKeyEntry()).build();
+        try (ZipFile in = new ZipFile(unsignedJarPath.toFile());
+                FileOutputStream out = new FileOutputStream(signedJarPath.toFile())) {
+            signer.sign(in, out);
+        }
+        JarResultBuildStep.filterJarFile(signedJarPath, unsignedJarPath, Set.of("java/lang/Integer.class"));
+        try (JarFile jarFile = new JarFile(unsignedJarPath.toFile())) {
+            jarFile.stream().map(JarEntry::getName).forEach(System.out::println);
```
Comment (gastaldi) [inline core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java:62]: This can be removed

**Review by gastaldi — CHANGES_REQUESTED** [review]
Added some comments. Can you please squash your commits?

File: core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java [hunk 3]
```diff
@@ -2,27 +2,63 @@
...
 class JarResultBuildStepTest {
 
     @Test
-    void should_unsign_jar_when_filtered(@TempDir Path tempDir) throws Exception {
-        Path signedJarFilePath = Path.of(getClass().getClassLoader().getResource("signed.jar").toURI());
-        Path jarFilePath = tempDir.resolve("unsigned.jar");
-        JarResultBuildStep.filterJarFile(signedJarFilePath, jarFilePath,
-                Set.of("org/eclipse/jgit/transport/sshd/SshdSessionFactory.class"));
-        try (JarFile jarFile = new JarFile(jarFilePath.toFile())) {
+    void should_unsign_jar_when_filtered(@TempDir Path tmp) throws Exception {
```
Comment (gastaldi) [inline core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java:49]: Can you keep the same parameter name? That will make it easier to notice what really changed here

File: core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java [hunk 4]
```diff
@@ -31,4 +67,40 @@ void should_unsign_jar_when_filtered(@TempDir Path tempDir) throws Exception {
...
+    private static Certificate createCertificate(KeyPair keyPair, String subjectDN)
+            throws OperatorCreationException, CertificateException, CertIOException {
+        Provider bcProvider = new BouncyCastleProvider();
+        Security.addProvider(bcProvider);
+        long now = System.currentTimeMillis();
+        Date startDate = new Date(now);
+        X500Name dnName = new X500Name(subjectDN);
+        BigInteger certSerialNumber = new BigInteger(Long.toString(now));
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(startDate);
```
Comment (gastaldi) [inline core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java:89]: AFAIK a Calendar is already initialized with the current time, why are you setting the time here?

**Comment by DShivansh** [comment]
@gastaldi  can you please re-review it?

**Review by gastaldi — CHANGES_REQUESTED** [review]
The code looks okay, but you need to squash all your commits

**Comment by DShivansh** [comment]
Hey @gastaldi 
I think I have messed something in git and I am not able to fix the history
can I create another pull request?

**Comment by gastaldi** [comment]
You can simply force push `git push -f` to your `jar-on-the-fly` branch and this PR will be updated automatically

**Comment by DShivansh** [comment]
force push from the new branch?

**Comment by DShivansh** [comment]
@gastaldi I have squashed all the commits.

**Review by gastaldi — APPROVED** [review]
LGTM now, good job!

**Comment by gastaldi** [comment]
@DShivansh you need to build the sources and force push again with `mvn process-sources `  to fix the formatting issues

**Comment by gastaldi** [comment]
Thanks!

**Comment by galderz** [comment]
Thanks @DShivansh!

**Pull request merged**